### PR TITLE
Some handle related improvements

### LINKED ISSUE #207: Implement one-to-one mapping of discord user and handle

### PR DESCRIPTION
**Pull request by meooow25**
- Make user to handle mapping obey a guild boundary instead of being global. This fixes things like `;handle list` and `;plot distrib` using data across servers.
- Ensure one-to-one discord user to handle mapping for a guild.
- Also add a lot more fields to the Codeforces User class so that we can do cool stuff in the future.
- Clean up old leftover code.

Resolves #207.

### CODE REVIEW

**Comment by meooow25** [comment]
On hold until #195 is merged to handle db changes at once.

**Comment by krofna** [comment]
Can you update this PR? :eyes:

**Comment by meooow25** [comment]
Updated. Ready for review now.

File: tle/cogs/duel.py [hunk 1]
```diff
@@ -3,15 +3,20 @@
...
+
 from tle.util.db.user_db_conn import Duel, Winner
 from tle.util import codeforces_api as cf
 from tle.util import codeforces_common as cf_common
 from tle.util import paginator
 from tle.util import discord_common
 from tle.util import table
 
+# TODO: This is a temporary workaround. Make Duel operate with guild boundaries.
+_DUEL_GUILD_ID = int(environ.get('DUEL_GUILD_ID'))
```
Comment (krofna) [inline tle/cogs/duel.py]: waturr

Comment (meooow25) [reply tle/cogs/duel.py]: That's basically how it is right now... this is just being explicit 👀

File: tle/cogs/duel.py [hunk 2]
```diff
@@ -29,7 +34,7 @@ def elo_delta(player, opponent, win):
 
 def get_cf_user(userid):
     handle = cf_common.user_db.gethandle(userid)
-    return cf_common.user_db.fetch_cfuser(handle)
+    return cf_common.user_db.fetch_cf_user(handle, _DUEL_GUILD_ID)
```
Comment (krofna) [inline tle/cogs/duel.py]: fetch_cf_user doesn't take guild id and it doesn't make sense either; it just fetches from cache

Comment (meooow25) [reply tle/cogs/duel.py]: Mistake 😬

**Pull request merged**